lib/transaction/transaction.js

```diff
@@ -205,6 +205,10 @@ Transaction.prototype._isInvalidSatoshis = function() {
 };
 
 Transaction.prototype._hasFeeError = function(opts) {
+  if (this._getUnspentValue() < 0) {
```

Comment:
As it stands now `_getUnspentValue()` is called around five times for the serialization check, would be interesting to see if this problematic with a transaction with many inputs/outputs.